packages/atomic/src/components/commerce/product-template-components/atomic-product-image/atomic-product-image.tsx

```diff
@@ -0,0 +1,153 @@
+import {Product, ProductTemplatesHelpers} from '@coveo/headless/commerce';
+import {Component, h, Prop, Element, State, Method} from '@stencil/core';
+import {Bindings} from '../../../../components';
+import {
+  InitializeBindings,
+  InitializableComponent,
+} from '../../../../utils/initialization-utils';
+import {filterProtocol} from '../../../../utils/xss-utils';
+import {Carousel} from '../../../common/carousel';
+import {ProductContext} from '../product-template-decorators';
+
+/**
+ * The `atomic-product-image` component renders an image from a product field.
+ */
+@Component({
+  tag: 'atomic-product-image',
+  styleUrl: 'atomic-product-image.pcss',
+  shadow: false,
+})
+export class AtomicProductImage implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @ProductContext() private product!: Product;
+  @Element() private host!: HTMLElement;
+  @State() private useFallback = false;
+  @State() private currentImage = 0;
+
+  /**
+   * The product field which the component should use. This will look for the field in the product object first, then in the product.additionalFields object.
+   */
+  @Prop({reflect: true}) field: string = 'ec_thumbnails';
```

Comment:
Potentially might want to support (at a later time) the ability to click on the image to open a model with high resolution `ec_images`...